File: galaxy_restaurant/app/models/order_item.rb
class OrderItem < ApplicationRecord
  belongs_to :order
  belongs_to :item

  validates :quantity, presence: true
  validates :status, presence: true
  validates :status, inclusion: { in: %w(ordered preparing ready_to_serve delivered) }

  enum :status, { ordered: 0, preparing: 1, ready_to_serve: 2, delivered: 3 }

  def price
    item.price * quantity
  end
end

File: server/src/models/order_item.py
import sqlalchemy as sa
from sqlalchemy.orm import relationship

from src.models._meta import Base


class OrderItem(Base):
    __tablename__ = "order_items"

    order_id = sa.Column(
        sa.String,
        sa.ForeignKey("orders.id"),
        index=True,
        nullable=False,
    )
    order = relationship("Order", back_populates="order_items")

    menu_item_id = sa.Column(
        sa.String,
        sa.ForeignKey("menu_items.id"),
        index=True,
        nullable=False,
    )
    menu_item = relationship("MenuItem", back_populates="order_items")

    quantity = sa.Column(sa.Integer, nullable=False)
    status = sa.Column(
        sa.String,
        sa.CheckConstraint(
            "status in ( 'ordered', 'preparing', 'ready_to_serve', 'delivered' )"
        ),
        nullable=False,
        server_default="ordered",
    )
